orttraining/orttraining/test/training_api/checkpoint/checkpoint_test.cc

```diff
@@ -203,7 +203,15 @@ TEST(CheckpointApiTest, SaveOptimizerStateAsCheckpoint_ThenLoad_CPU) {
     auto param = std::make_shared<Parameter>(it->first, it->second, is_trainable);
     named_parameters.insert({it->first, param});
   }
-  auto optimizer = Optimizer(model_uri, named_parameters);
+
+  onnxruntime::SessionOptions session_option;
+  std::unique_ptr<Environment> env;
+  ORT_THROW_IF_ERROR(Environment::Create(nullptr, env));
+  auto optim_sess = std::make_unique<onnxruntime::InferenceSession>(session_option, *env);
+  ORT_THROW_IF_ERROR(optim_sess->Load(model_uri));
+  ORT_THROW_IF_ERROR(optim_sess->Initialize());
+
+  auto optimizer = Optimizer(named_parameters, optim_sess.get());
```

Comment:
What is the plan here wrt the C APIs that @askhade is working on? Can we possibly pass the session options and env to Optimizer/Module and let them create the sessions in the internal APIs, and C API can pass in the centralized session options/env?

Reply:
Put the session management back to Module/Optimizer, making Askhade's change easier to iterate. 

This PR explored what APIs and ways we expose APIs at minimum efforts so far. We need more work making those APIs real ORT-styple C/C++ public APIs, which will be Interated in Askhade's changes.